modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -131,20 +146,22 @@ def eval_host(ip, share, subdir = '')
 
     return read, write, msg, nil if skip
 
-    rfd = simple.client.find_first("#{subdir}\\*")
-    read = true if !rfd.nil?
+    if Connected
+      tree = simple.client.last_tree
+    else
+      tree = simple.client.tree_connect("\\\\#{ip}\\#{share}")
+    end
```

Comment:
We'd want to avoid using constants to keep track of module state. This value isn't set to `false`/`true`, so it might have other issues too 👀 

It might be easier to connect once to iterate folders/files